src/System.Text.Json/tests/Serialization/ExtensionDataTests.cs

```diff
@@ -353,21 +354,129 @@ private class ClassWithReference
             public ClassWithExtensionProperty MyReference { get; set; }
         }
 
+        [Theory]
+        [InlineData(@"{""MyIntMissing"":2,""MyReference"":{""MyIntMissingChild"":3}}")]
+        [InlineData(@"{""MyReference"":{""MyIntMissingChild"":3},""MyIntMissing"":2}")]
+        [InlineData(@"{""MyReference"":{""MyNestedClass"":null,""MyInt"":0,""MyIntMissingChild"":3},""MyIntMissing"":2}")]
+        public static void NestedClass(string json)
+        {
+            ClassWithReference obj;
+
+            void Verify()
+            {
+                Assert.IsType<JsonElement>(obj.MyOverflow["MyIntMissing"]);
+                Assert.Equal(1, obj.MyOverflow.Count);
+                Assert.Equal(2, obj.MyOverflow["MyIntMissing"].GetInt32());
+
+                ClassWithExtensionProperty child = obj.MyReference;
+
+                Assert.IsType<JsonElement>(child.MyOverflow["MyIntMissingChild"]);
+                Assert.IsType<JsonElement>(child.MyOverflow["MyIntMissingChild"]);
+                Assert.Equal(1, child.MyOverflow.Count);
+                Assert.Equal(3, child.MyOverflow["MyIntMissingChild"].GetInt32());
+                Assert.Null(child.MyNestedClass);
+                Assert.Equal(0, child.MyInt);
+            }
+
+            obj = JsonSerializer.Deserialize<ClassWithReference>(json);
+            Verify();
+
+            // Round-trip the json and verify.
+            json = JsonSerializer.Serialize(obj);
+            obj = JsonSerializer.Deserialize<ClassWithReference>(json);
+            Verify();
+        }
+
+        private class ParentClassWithObject
+        {
+            public string Text { get; set; }
+            public ChildClassWithObject Child { get; set; }
+
+            [JsonExtensionData]
+            public Dictionary<string, object> ExtensionData { get; set; } = new Dictionary<string, object>();
+        }
+
+        private class ChildClassWithObject
+        {
+            public int Number { get; set; }
+
+            [JsonExtensionData]
+            public Dictionary<string, object> ExtensionData { get; set; } = new Dictionary<string, object>();
```

Comment:
Can we add a test for a custom `IDictionary<string, object>` that we put the `JsonExtensionData` on, if one doesn't exist? Or even use the built-in ones like `SortedDictionary<TKey,TValue>` other than just regular Dictionary.

Reply:
There are tests that already verify IDictionary (any test that uses class `ClassWithExtensionProperty`). However I'll change one of the test classes to use `SortedDictionary` intead of `Dictionary` just in case...